File: javascript/sql_injection.js
const sqlite3 = require('better-sqlite3');
const readline = require('readline');

const db = new sqlite3(':memory:');
db.exec('CREATE TABLE users (name TEXT, password TEXT)');
db.exec("INSERT INTO users VALUES ('alice','wonderland'),('bob','builder')");

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
rl.question('User: ', (username) => {
  // CWE-89: user input interpolated directly into SQL query.
  const rows = db.prepare(`SELECT * FROM users WHERE name = '${username}'`).all();
  console.log(rows);
  rl.close();
});

File: python/sql_injection.py
import sqlite3

conn = sqlite3.connect(':memory:')
c = conn.cursor()
c.execute('CREATE TABLE users (name TEXT, password TEXT)')
c.executemany('INSERT INTO users VALUES (?, ?)', [
    ('alice', 'wonderland'),
    ('bob', 'builder'),
])

username = input('User: ')
# CWE-89: attacker-controlled input interpolated directly into SQL query.
query = f"SELECT * FROM users WHERE name = '{username}'"
for row in c.execute(query):
    print(row)
